### PR TITLE
[FLINK-20078][coordination] Factor out an ExecutionGraph factory method for DefaultExecutionTopology

### PR DESCRIPTION
**Pull request by azagrebin**
follow-up based on #13958 and discussion https://github.com/apache/flink/pull/13958#discussion_r519797327.
only last commit is to review.

### CODE REVIEW

**Comment by azagrebin** [comment]
cc @zhuzhurk @tillrohrmann

**Review by tillrohrmann — APPROVED** [review]
Thanks for creating this PR @azagrebin. LGTM. +1 for merging.

File: flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adapter/DefaultExecutionTopology.java [hunk 1]
```diff
@@ -140,6 +115,53 @@ public DefaultSchedulingPipelinedRegion getPipelinedRegionOfVertex(final Executi
 		return pipelinedRegion;
 	}
 
+	public static DefaultExecutionTopology fromExecutionGraph(ExecutionGraph executionGraph) {
+		checkNotNull(executionGraph, "execution graph can not be null");
+
+		ExecutionGraphIndex executionGraphIndex = computeExecutionGraphIndex(
+			executionGraph.getAllExecutionVertices(),
+			executionGraph.getNumberOfExecutionJobVertices());
```
Comment (zhuzhurk) [inline flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adapter/DefaultExecutionTopology.java:123]:
```suggestion
			executionGraph.getTotalNumberOfVertices());
```

**Review by zhuzhurk — APPROVED** [review]
Thanks for addressing the comment. +1 for merging once CI givens green.

**Pull request closed without merging**